clang/lib/Lex/LiteralSupport.cpp

```diff
@@ -1432,6 +1462,42 @@ NumericLiteralParser::GetFloatValue(llvm::APFloat &Result) {
                                                : APFloat::opInvalidOp;
 }
 
+bool NumericLiteralParser::GetDecimalFloatValue(llvm::APDecimalFloat &Result) {
+  using llvm::APFloat;
+
+  unsigned n = std::min(SuffixBegin - ThisTokBegin, ThisTokEnd - ThisTokBegin);
+
+  llvm::SmallString<16> Buffer;
+  StringRef Str(ThisTokBegin, n);
+  // if (Str.contains('\'')) {
+  Buffer.reserve(n);
+  std::remove_copy_if(Str.begin(), Str.end(), std::back_inserter(Buffer),
+                      &isDigitSeparator);
+  Str = Buffer;
+  //}
+
+  BID_UINT32 result;
+  BID_UINT32 result2;
+  unsigned int x = 0;
+  printf("%s\n", Buffer.c_str());
+  result2 = bid32_from_string((char *)Buffer.c_str(), x, &result);
+  printf("%08x\n", result2);
+
+  llvm::APInt store(32u, result2, false);
+
+  Result = llvm::APDecimalFloat(
+      store, llvm::decFltSemantics(32u, /*Scale=*/1u, /*IsSigned=*/0u,
+                                   /*IsSaturated=*/false,
+                                   /*HasUnsignedPadding=*/false));
```

Comment:
This will presumably need to be generalized to handle 64-bit and 128-bit literals.